### ISSUE TITLE
Should we only deploy to vercel if CI passes?

### ISSUE THREAD

**Issue by masonmcelvain** [iFixit/react-commerce#1811]
Should we only publish production deployments on Vercel when pulls merge if our CI passes? Is this even something Vercel allows us to do?

See https://ifixit.slack.com/archives/C8KBFH1JT/p1688763856750959?thread_ts=1688761819.993379&cid=C8KBFH1JT

**Comment by masonmcelvain** [comment]
https://github.com/orgs/vercel/discussions/76

https://vercel.com/guides/how-can-i-use-gitlab-pipelines-with-vercel

**Comment by masonmcelvain** [comment]
## GitHub Actions

- [Docs](https://vercel.com/guides/how-can-i-run-end-to-end-tests-after-my-vercel-preview-deployment) on testing the deployment directly, instead of setting up a local instance.
- There doesn't seem to be a way to conditionally deploy to vercel with their GitHub actions deploy tool.
  - We could make a [custom deploy workflow](https://github.com/orgs/vercel/discussions/2294) with the vercel cli
    - [Another example](https://aaronfrancis.com/2021/the-perfect-vercel-github-actions-deployment-pipeline) of this

## Vercel Checks

Alternatively, we can make our own ["Vercel Checks"](https://vercel.com/docs/concepts/observability/checks-overview) integration that tests the deployment directly, and can prevent the www.ifixit.com domain from being assigned if the checks fail.

It seems like we would set up a webhook responder that runs checks (e.g. Playwright) against the deployment, and [responds](https://vercel.com/docs/concepts/observability/checks-overview/creating-checks#body-attributes) to Vercel's API with whether the checks were successful.

---

@iFixit/qae Can I get your opinions on ways to approach this problem given the tools available to us?

**Comment by dhmacs** [comment]
What prompted this? What's the reason we want to prevent Vercel from deploying if CI fails?

**Comment by masonmcelvain** [comment]
@jarstelfox [suggested](https://ifixit.slack.com/archives/C8KBFH1JT/p1688763856750959?thread_ts=1688761819.993379&cid=C8KBFH1JT) this idea in Slack when we were investigating a Strapi deploy issue. Turns out that preventing Vercel deploys wouldn't have helped in that case, but we wrote down the idea because we follow this pattern in our PHP app. If CI fails during a deploy of our PHP app, then we roll back to the most recent deployment.

That said, @sterlinghirsh and I couldn't think of a case where react-commerce CI caught a bug that should have prevented a Vercel deploy.

Additionally, our playwright tests are still relatively flaky. If CI failures prevented Vercel deploys, then flaky tests would require manual intervention to publish a deploy.

We will eventually have to reconsider our Vercel deploy process when we pull react-commerce into the iFixit monorepo. I think we can close this issue for now because the feature will likely be more work than it is worth, and perhaps reevaluate when we migrate to the monorepo.
